src/app/backend/handler/apihandler.go

```diff
@@ -220,22 +243,22 @@ func CreateHTTPAPIHandler(client *clientK8s.Clientset, heapsterClient client.Hea
 
 	apiV1Ws.Route(
 		apiV1Ws.POST("/appdeployment").
-			To(apiHandler.handleDeploy).
+			To(apiHandler.injectApiClient(apiHandler.handleDeploy)).
```

Comment:
The shared in cluster user must have only access to the kubernetes api version in order for the dashboard to start. I think we should try to get rid of it entirely if it's run behind a reverse proxy.

With the current logic you can also just not set the bearer token and you will use the shared user. Maybe it makes sense to add a cli flag to disable this behaviour.

>Agreed maybe modify the APIHandler client to take a method?

Sounds reasonable.